pkg/process/scheduler/exclusive.go

```diff
@@ -0,0 +1,107 @@
+package scheduler
+
+import (
+	"context"
+	"os"
+	"sync"
+
+	"github.com/leptonai/gpud/pkg/log"
+	"github.com/leptonai/gpud/pkg/process"
+)
+
+var _ Scheduler = &exclusiveScheduler{}
+
+func NewExclusiveScheduler() Scheduler {
+	return &exclusiveScheduler{}
+}
+
+// exclusiveScheduler is a scheduler that runs a single process at a time.
+// Does not support concurrent execution of multiple processes.
+type exclusiveScheduler struct {
+	mu             sync.RWMutex
+	runningProcess process.Process
+}
+
+func (s *exclusiveScheduler) CountProcesses() (int, error) {
```

Comment:
We defined it but no one used it.. We should consider if it is necessary.. If not, we should remove it.